.github/workflows/test.yml

```diff
@@ -104,8 +118,6 @@ jobs:
 
     # this one runs as root for CAP_SETUID
     - name: test run-as-user
-      env:
-        TEST_SUBPROC_USER: testbot
```

Comment:
this is testing the ability to change users. it passes without TEST_SUBPROC_USER, but it's a weaker test

I wrote this originally -- you're welcome to merge as-is + I can figure out user creation afterwards. I made a follow up ticket [here](https://viam.atlassian.net/browse/APP-3946)

Reply:
note the skip on [this line](https://github.com/viamrobotics/goutils/actions/runs/7893349207/job/21541805668#step:15:22)
> managed_process_test.go:197: skipping run-as-user because setuid required elevated privileges

Reply:
ah, thanks I missed this - I will try to add another user and run the test as it was intended. if it's too cumbersome then yeah, i'll just make a reference to the ticket as a TODO

Reply:
yeah, made a few simple attempts but no luck... I added the TODO with a link to this thread.